lib/util/object.ts

```diff
@@ -11,3 +11,11 @@ export function hasKey<K extends string, T>(
 ): o is T & Record<K, unknown> {
   return typeof o === 'object' && k in o;
 }
+
+export function filterUndefined(input: unknown): void {
+  for (const [key, val] of Object.entries(input)) {
+    if (val === undefined) {
+      delete input[key]; // eslint-disable-line no-param-reassign
+    }
+  }
+}
```

Comment:
What about this?

```typescript
function filterUndefined(input: any): any {
  if (is.array(input)) {
    return input.filter((x) => x !== undefined).map((x) => filterUndefined(x));
  }

  if (is.object(input)) {
    const ret: any = {};
    for (const [key, value] of Object.entries(input)) {
      if (value !== undefined) {
        ret[key] = filterUndefined(value);
      }
    }
    return ret;
  }
  return input;
}
```

Works for arrays and object, filters recursively, avoids angering ESLint :joy: